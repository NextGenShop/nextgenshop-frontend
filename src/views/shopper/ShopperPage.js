import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import Container from "@material-ui/core/Container";
import Grid from "@material-ui/core/Grid";
import ProductCatalog from "../../components/ProductCatalog";
import ProductSearchBox from "../../components/ProductSearchBox";
import RetailerSelect from "../../components/RetailerSelect";
import mockBasketItems from "../../store/mock/MockBasketItems.json";
import ArtificialCashier from "../../components/ArtificialCashier";
import Basket from "../../components/Basket";
import { addBasketItem, removeBasketItem } from "../../utils/basketUtils";

const useStyles = makeStyles((theme) => ({
  root: {
    paddingTop: theme.spacing(2),
    paddingBottom: theme.spacing(2),
  },
  mt: {
    marginTop: theme.spacing(3),
  },
}));

export default function ShopperPage() {
  const classes = useStyles();

  const [items, setItems] = React.useState(mockBasketItems);
  const [searchQuery, setSearchQuery] = React.useState("");
  const [retailer, setRetailer] = React.useState("Tesco");

  return (
    <Container maxWidth="xl" className={classes.root}>
      <Grid container spacing={3}>
        <Grid item md={6}>
          <Typography variant="subtitle1">Artificial Cashier</Typography>
        </Grid>
        <Grid item xs={6}>
          <Typography variant="subtitle1">Product Catalogue</Typography>
        </Grid>
      </Grid>
      <Grid container spacing={3}>
        <Grid item md={6}>
          <ArtificialCashier />
        </Grid>
        <Grid item xs={6}>
          <Grid container spacing={3}>
            <Grid item xs={6}>
              <ProductSearchBox
                value={searchQuery}
                onChange={(e) => {
                  setSearchQuery(e.target.value);
                }}
              />
            </Grid>
            <Grid item xs={6}>
              <RetailerSelect
                value={retailer}
                onChange={(event, newValue) => {
                  setRetailer(newValue);
                }}
              />
            </Grid>
          </Grid>
          <ProductCatalog
            retailer={retailer}
            addToBasket={(product) => {
              addBasketItem(product, items, setItems);
            }}
            searchQuery={searchQuery}
            limit={9}
          />
          <hr className={classes.mt} />
          <Typography variant="subtitle1">Shopping Basket</Typography>
          <Basket
            items={items}
            removeItem={(basketItemId) => removeBasketItem(basketItemId, items, setItems)}
          />
        </Grid>
      </Grid>
    </Container>
  );
}
