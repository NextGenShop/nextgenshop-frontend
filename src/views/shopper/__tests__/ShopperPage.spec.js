import React from 'react';
import { shallow } from 'enzyme';
import { ShopperPage } from '../ShopperPage';
import mockUsers from '../../../store/mock/MockUsers.json';

describe('ShopperPage', () => {
  test('should correctly render with props', () => {
    const component = shallow(<ShopperPage authUser={mockUsers[0]} />);
    expect(component).toMatchSnapshot();
  });
});
