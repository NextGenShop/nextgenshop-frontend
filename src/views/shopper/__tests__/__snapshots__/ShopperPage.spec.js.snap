// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShopperPage should correctly render with props 1`] = `
<WithStyles(ForwardRef(Container))
  className="makeStyles-root-1"
  maxWidth="xl"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={6}
    >
      <WithStyles(ForwardRef(Typography))
        variant="subtitle1"
      >
        Artificial Cashier
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={6}
    >
      <WithStyles(ForwardRef(Typography))
        variant="subtitle1"
      >
        Product Catalogue
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={6}
    >
      <withRouter(Connect(ArtificialCashier)) />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={6}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        spacing={3}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <ProductSearchBox
            onChange={[Function]}
            value=""
          />
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <Connect(ProductCatalog)
        limit={9}
        retailer="Mock Retailer"
        searchQuery=""
        shopperId={1}
      />
      <hr
        className="makeStyles-mt-2"
      />
      <WithStyles(ForwardRef(Typography))
        variant="subtitle1"
      >
        Shopping Basket
      </WithStyles(ForwardRef(Typography))>
      <Connect(Basket)
        shopperId={1}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Container))>
`;
